File: app/src/main/java/ru/alexsuvorov/paistewiki/Fragment/CymbalsFragment.java
package ru.alexsuvorov.paistewiki.Fragment;

import android.content.res.TypedArray;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;

import java.util.ArrayList;

import ru.alexsuvorov.paistewiki.Adapter.CymbalChoiseAdapter;
import ru.alexsuvorov.paistewiki.ItemModel.Product;
import ru.alexsuvorov.paistewiki.R;

import static ru.alexsuvorov.paistewiki.R.id.container;

public class CymbalsFragment extends Fragment implements CymbalChoiseAdapter.OnProductSelected {

    ArrayList<Product> products = new ArrayList<Product>();
    CymbalChoiseAdapter boxAdapter;
    Fragment fragment = null;
    Class fragmentClass = null;

    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.cymbals_fragment, container, false);

        // создаем адаптер
        products.clear();
        fillData();
        boxAdapter = new CymbalChoiseAdapter(getContext(), products, this); //this - интерфейс


        // настраиваем список
        ListView lvMain = (ListView) view.findViewById(R.id.cymbalsList);
        lvMain.setAdapter(boxAdapter);
        return view;

    }

    // генерируем данные для адаптера
    void fillData() {
        for (int i = 0; i <= 17; i++) {
            String[] cymbalSeriesNamesArr = getResources().getStringArray(R.array.cymbalSeriesNames);
            //String[] cymbalSeriesSubNamesArr = getResources().getStringArray(R.array.cymbalSeriesSubNames);

            //TypedArray cymbalPicsArr = getResources().obtainTypedArray(R.array.cymbalPics);
            TypedArray cymbalSeriesPicsArr = getResources().obtainTypedArray(R.array.cymbalSeriesPics);

            products.add(new Product(cymbalSeriesNamesArr[i], null,
                    cymbalSeriesPicsArr.getDrawable(i), null)); //

            cymbalSeriesPicsArr.recycle();
        }

    }

    @Override
    public void onResume() {
        super.onResume();

        getActivity().setTitle(R.string.nav_header_cymbalsbutton);
    }

    //Когда нажали на кнопку , адаптер вернет в функцию onProductSelected
    //которая находится во фрагменте, позицию товара в листе которую выбрали,
    //то есть в фрагменте мы будем знать что выбрали
    //и сможем передать только один продукт в следующий экран

    @Override
    public void onProductSelected(int productIndex) {
        //Product selectedProduct = products.get(productIndex);

        fragmentClass = SeriesDescriptionFragment.class;

        try {
            fragment = (Fragment) fragmentClass.newInstance();
        } catch (Exception e) {
            e.printStackTrace();
        }

        SeriesDescriptionFragment SerDescr = new SeriesDescriptionFragment();
        Bundle bundle = new Bundle();
        bundle.putInt("productIndex", productIndex);
        SerDescr.setArguments(bundle);
        FragmentManager fragmentManager = getFragmentManager();
        fragmentManager.beginTransaction().addToBackStack(null).replace(container, SerDescr).commit();
    }

}

File: feature/product_list/impl/src/main/java/ru/alexsuvorov/paistewiki/feature/product_list/CymbalsFragment.kt
package ru.alexsuvorov.paistewiki.feature.product_list

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import ru.alexsuvorov.paistewiki.core.database.AppDatabase
import ru.alexsuvorov.paistewiki.core.database.dao.CymbalDao
import ru.alexsuvorov.paistewiki.core.database.model.CymbalSeries
import ru.alexsuvorov.paistewiki.feature.product_list.adapter.CymbalsAdapter
import ru.alexsuvorov.paistewiki.tools.AppPreferences

class CymbalsFragment : Fragment() {

    private var cymbalSeries: List<CymbalSeries> = ArrayList()
    private var cymbalsAdapter: CymbalsAdapter? = null
    private var db: AppDatabase? = null
    private var cymbalDao: CymbalDao? = null
    private var cymbalsView: RecyclerView? = null
    private var appPreferences: AppPreferences? = null

    interface CymbalsNavigation {
        fun onCymbalSeriesSelected(seriesId: Int?)
    }

    private var navigation: CymbalsNavigation? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is CymbalsNavigation) {
            navigation = context
        }
        appPreferences = AppPreferences(requireContext())
    }

    override fun onDetach() {
        super.onDetach()
        navigation = null
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_cymbals, container, false)
        cymbalsView = view.findViewById(R.id.cymbalsList)
        cymbalsView?.isNestedScrollingEnabled = false
        cymbalsView?.setHasFixedSize(false)
        cymbalsView?.layoutManager = LinearLayoutManager(activity)
        db = context?.let { AppDatabase.getDatabase(it) }
        cymbalDao = db?.cymbalDao()
        cymbalSeries = cymbalDao?.getAllProduced(1) ?: ArrayList()
        cymbalsAdapter = CymbalsAdapter(cymbalSeries, requireContext(), object : CymbalsAdapter.OnItemClickListener {
            override fun onItemClick(cymbalSeries: CymbalSeries?) {
                navigation?.onCymbalSeriesSelected(cymbalSeries?.cymbalseries_id)
            }
        })

        cymbalsView?.adapter = cymbalsAdapter
        return view
    }
}
